### ISSUE TITLE
[feaure request] [usability] improve access value editor

### ISSUE THREAD

**Issue by Locke** [openstreetmap/iD#3945]
A user struggled to update access tags, as "motorcar" isn't visible in the upper part of the editor.

![screenshot from 2017-04-02 12 57 58_](https://cloud.githubusercontent.com/assets/53905/24586714/7cffdcb8-17a7-11e7-982a-26e57f3bdbc3.png)

Initially motor_vehicle=agricultural and motorcar=no were set. As he couldn't see motorcar in the upper editor he couldn't change it. (In later attempts he added access=yes, foot=yes, motor_vehicle=yes, bicycle=yes and horse=yes which all had no effect to the more specific tag motorcar).

I suggest to add more possible tags, e.g. everything from https://wiki.openstreetmap.org/wiki/Key:access#Transport_mode_restrictions . As just adding everything would be too much I think most keys should be hidden with a (+) icon which opens a tree.

Also, the default (grayed out "yes") needs to be updated depending on higher access tags: When a way has vehicle=no motor_vehicle=destination the default value of bicycle is still displayed as grayed out "yes", however it should be "no" as bicycle is included in vehicle. As I see it from #2213 only the value of "access" is inherited and not "vehicle".

**Comment by freakyuser-osm** [comment]
I think hgv should be added as an preset to the access presets in id. This way it's easier for beginners to map truck access restrictions.

(See: https://github.com/openstreetmap/iD/issues/3946)

**Comment by bhousel** [comment]
> I suggest to add more possible tags, e.g. everything from https://wiki.openstreetmap.org/wiki/Key:access#Transport_mode_restrictions . As just adding everything would be too much I think most keys should be hidden with a (+) icon which opens a tree.

I like this suggestion because it would let us support more keys, and at the same time we could drop the existing keys for things like `horse` `bicycle` `foot` that are almost never used and just take up space.



> Also, the default (grayed out "yes") needs to be updated depending on higher access tags: When a way has vehicle=no motor_vehicle=destination the default value of bicycle is still displayed as grayed out "yes", however it should be "no" as bicycle is included in vehicle. As I see it from #2213 only the value of "access" is inherited and not "vehicle".

Yes, I never heard of `vehicle` until just now.

**Comment by slhh** [comment]
I think the access field should show a complete list of all present access tags of the feature. This should include any ":conditional", ":lanes", ":forward", or ":backward" key modifiers or their combinations, because we should not display misleading incomplete information.
This list can be a read-only raw tag listing, which is more condensed than the all tags section. Clicking anywhere on the list should unfold the access editing tree as suggested above.

>Also, the default (grayed out "yes") needs to be updated depending on higher access tags
Yes, and even higher level conditional settings would change the default. We do not need to evaluate the default in this case, but we can indicate it as "conditional".

In addition country and highway type specific defaults needs to be considered (https://wiki.openstreetmap.org/wiki/OSM_tags_for_routing/Access-Restrictions). In case we don't want to make the behavior country specific, we can display "country default" or "???"  instead of a clear (but maybe wrong) "yes" or "no" as the grey default value where required.

>for things like horse bicycle foot that are almost never used

I wouldn't call it almost never used, there are 750000 uses of horse, and nearly 4 milllion uses of foot and bicycle each.
> I never heard of vehicle until just now.

It's in the wiki since 2008, and there are 152000 uses plus 13000 uses of vehicle:conditional.

**Comment by SelfishSeahorse** [comment]
It would be nice if at least the `vehicle` access key would be supported, because the traffic sign 'closed to all vehicles in both directions' seems to be one of the most used restrictive signs (at least in Europe).

I'm unsure if adding another access category to *Allowed Access* wouldn't make this section too confusing, but maybe iD could support this key in the background, i.e.:

- If there's already a `vehicle` key, its value could appear next to *Motor Vehicles* and *Bicycles*.
- If there's no `vehicle` key and someone enters the same value to *Motor Vehicles* and *Bicycles*, the `motor_vehicle=*` and `bicycle=*` tag could be be replaced by a `vehicle=*` tag.
- If there's already a `vehicle` key and someone changes either *Motor Vehicle* or *Bicycle* access, the `vehicle=*` tag could be replaced by `motor_vehicle=*` and `bicycle=*` keys (with the value of `vehicle=*` being copied to the key that hasn't been changed).

**Comment by matkoniecz** [comment]
> If there's already a vehicle key, its value could appear next to Motor Vehicles and Bicycles

this would be highly useful, too often people using iD add useless `motor_vehicle` tags where `vehicle` tag is set already.

> If there's no vehicle key and someone enters the same value to Motor Vehicles and Bicycles, the motor_vehicle=* and bicycle=* tag could be be replaced by a vehicle=* tag.

I am not sure is it the best idea, it seems to be a bit too smart.

> If there's already a vehicle key and someone changes either Motor Vehicle or Bicycle access, the vehicle=* tag could be replaced by motor_vehicle=* and bicycle=* keys (with the value of vehicle=* being copied to the key that hasn't been changed).

What about other vehicles?

**Comment by SelfishSeahorse** [comment]
>> If there's already a vehicle key and someone changes either Motor Vehicle or Bicycle access, the vehicle=* tag could be replaced by motor_vehicle=* and bicycle=* keys (with the value of vehicle=* being copied to the key that hasn't been changed).
> What about other vehicles?

I thought that bicycles are the only non-motor vehicles, but i've just noticed that there are also horse-drawn carriages (carriage=*), so this was a bad idea of mine. Maybe it's still best if a *Vehicle* entry would be added to *Allowed Access*.

**Comment by haukepribnow** [comment]
I've almost submitted a new issue - but now that I've found this discussion, I realize it would have been a duplicate. So therefore I'm posting my prepared issue here as a comment instead:

### `vehicle=no` does not yet entail `motor_vehicle=no`, `bicycle=no`, `carriage=no`

**Steps to reproduce**
Add `vehicle=no` to a highway in the iD editor sidebar.

**Expected behavior**
In the Access section of the iD editor sidebar, the entries for "Motor Vehicles" & "Bicycles" should show "no" in a grey color.

**Current behavior**
Both entries show "yes" in a grey color.

**Background**
[An advanced access hierarchy](https://wiki.openstreetmap.org/wiki/File:Access_hierarchy_simple2.png) making intense use of the `vehicle=` key was [approved in 2009](https://wiki.openstreetmap.org/wiki/Proposed_features/More_access_keys_and_values).

The core idea is: `vehicle=no` should entail `motor_vehicle=no`, `carriage=no`, `bicycle=no`

Despite the wide use of `vehicle=no` in Germany and despite tools like http://osmtools.de/traffic_signs/ that further propagate the use of `vehicle=no`, iD does not seem to implement this more advanced access hierarchy yet.

**Proposed fix**
Implement the simplified hierarchy here: https://wiki.openstreetmap.org/wiki/File:Access_hierarchy_simple2.png (I guess easier said than done.)

**Comment by haukepribnow** [comment]
Just had a quick look at the code to estimate the effort to tackle at least the `vehicle`-based hiearchy. The relevant code seems to be here: https://github.com/openstreetmap/iD/blob/a018d276eb0d28fc414e390fbd8c66b54c910d4c/modules/ui/fields/access.js#L228-L246

**Comment by impiaaa** [comment]
@1ec5 mentioned this issue to me, and I created a mockup for a more compact access UI, though I was not aware of the existing proposal for a tree view when I made it.
![access mockup](https://user-images.githubusercontent.com/1307275/206339405-9641aba9-2c16-4abf-9b83-0ba72f969e40.png)

**Comment by matkoniecz** [comment]
Field is partially defined in https://github.com/openstreetmap/id-tagging-schema/blob/3f445ce03496abd73802a9a4c3c66d65f8bd7571/data/fields/access.json

**Comment by tyrasd** [comment]
Yes, the access is a bit special (already have a special type `"type": "access",`), because its quite complex in how the access-tags are related to each other.

This is mostly a UI challenge to make it intuitive for mappers to use.
